lib/fs.js

```diff
@@ -406,7 +406,7 @@ function readFileAfterClose(err) {
   else
     buffer = context.buffer;
 
-  if (err) return callback(err, buffer);
+  if (err) return callback(err);
```

Comment:
Shouldn't this move to line 401, before the buffer object (which is no longer being passed), goes through the code to create/change it?